module-4/metaanalyse/README.md

```diff
@@ -0,0 +1,97 @@
+# PDF Metadata Extractor
+
+## Overview
+
+The **PDF Metadata Extractor** is a Python program designed to automate the extraction of metadata from PDF files. This program can process individual PDF files or entire directories containing multiple PDFs. The extracted metadata is saved to a CSV file, with fields clearly marked and empty fields denoted by `undefined`.
+
+## Features
+
+- **Extract Metadata**: Automatically extract metadata such as Title, Author, Creator, Created, Modified, Subject, Keywords, Description, Producer, and PDF Version from PDF files.
+- **Flexible Input**: Process a single PDF file or multiple files in a directory.
+- **CSV Output**: Save metadata to a CSV file with customizable filename and path.
+- **Clear Metadata Handling**: Metadata fields that are not present or empty in the PDF file are marked as undefined.
+
+## Requirements
+
+- Python 3.x
+- PyPDF2
+- argparse
+- csv
+- os
+
+## Installation
+
+Install the required dependencies using pip:
+
+```bash
+pip install PyPDF2
+```
```

Comment:
**for the requirements:** add a `requirements.txt` and adjust the install command to install requirements from the file.

**General:**
Add an overview about requirements and prerequisites, like e.g. a virtual environment and how to create one locally in order to isolate the package installations

Reply:
not for 2 items :D